src/sst/core/baseComponent.cc

```diff
@@ -535,188 +536,121 @@ uint8_t BaseComponent::getComponentInfoStatisticEnableLevel(const std::string& s
+      }
+  }
+
+  statistic->setRegisteredCollectionMode(statCollectionMode);
+}
+
+Statistics::StatisticBase*
+BaseComponent::createStatistic(Params& cpp_params, const Params& python_params,
+                               const std::string& name, const std::string& subId,
+                               StatCreateFunction fxn)
+{
+  auto* engine = Statistics::StatisticProcessingEngine::getInstance();
+
+  configureAllowedStatParams(cpp_params);
+  cpp_params.insert(python_params);
+  std::string type =  cpp_params.find<std::string>("type", "sst.AccumulatorStatistic");
+  auto* stat = fxn(this, engine, type, name, subId, cpp_params);
+  configureCollectionMode(stat, cpp_params, name);
+  engine->registerStatisticWithEngine(stat);
+
+  return stat;
+}
+
+Statistics::StatisticBase*
+BaseComponent::createEnabledAllStatistic(Params& params, const std::string &name, const std::string &statSubId,
+                                         StatCreateFunction fxn)
+{
+  for (auto* stat : m_enabledAllStats){
+    if (stat->getStatName() == name && stat->getStatSubId() == statSubId){
+      return stat;
     }
+  }
 
-    if (true == statGood) {
-        // Instantiate the Statistic here defined by the type here
-        //statistic = engine->createStatistic<T>(owner, statTypeParam, statName, statSubId, statParams);
-        statistic = create(statTypeParam, curr_info->component, statName, statSubId, statParams);
-        if (nullptr == statistic) {
-            out.fatal(CALL_INFO, 1, "ERROR: Unable to instantiate Statistic %s; exiting...\n", fullStatName.c_str());
-        }
-
-
-        // Check that the statistic supports this collection rate
-        if (false == statistic->isStatModeSupported(statCollectionMode)) {
-            if (StatisticBase::STAT_MODE_PERIODIC == statCollectionMode) {
-                out.verbose(CALL_INFO, 1, 0, " Warning: Statistic %s Does not support Periodic Based Collections; Collection Rate = %s\n", fullStatName.c_str(), collectionRate.toString().c_str());
-            } else {
-                out.verbose(CALL_INFO, 1, 0, " Warning: Statistic %s Does not support Event Based Collections; Collection Rate = %s\n", fullStatName.c_str(), collectionRate.toString().c_str());
-            }
-            statGood = false;
-        }
+  auto* stat = createStatistic(params, my_info->allStatConfig->params, name, statSubId, std::move(fxn));
+  m_enabledAllStats.push_back(stat);
+  return stat;
+}
 
-        // Tell the Statistic what collection mode it is in
-        statistic->setRegisteredCollectionMode(statCollectionMode);
-    }
+Statistics::StatisticBase*
+BaseComponent::createExplicitlyEnabledStatistic(Params &params, StatisticId_t id,
+                                                const std::string &name, const std::string &statSubId,
+                                                StatCreateFunction fxn)
+{
+  if (my_info->parent_info){
+    fatal(__LINE__, __FILE__, "createExplicitlyEnabledStatistic", 1,
```

Comment:
Why not just use the CALL_INFO or CALL_INFO long macro here instead of explicitly doing __LINE__, etc.?

Reply:
No reason in particular. Just didn't think to use it. We can change that to match other calls.

Reply:
:+1: